app/assets/javascripts/components/uploads/upload_viewer.jsx

```diff
@@ -1,181 +1,160 @@
+  const license = get(metadata, 'imageinfo[0].extmetadata.LicenseShortName.value');
+  const globalUsage = get(metadata, 'globalusage', []);
+  let usageTableElements;
+  if (globalUsage && pageViews !== undefined) {
+    usageTableElements = globalUsage.map((usage, index) => {
+        return (
+          <tr className="view-file-details" key={usage.url}>
+            <td className="row-details">{usage.wiki}&nbsp;&nbsp;&nbsp;</td>
+            <td className="row-details"><a href={usage.url}>{usage.title}</a>&nbsp;&nbsp;&nbsp;</td>
+            <td className="text-right row-details">{pageViews[index]}</td>
+          </tr>
+        );
     });
-  },
-
-  handleClickOutside() {
-    this.props.closeUploadViewer();
-  },
-
-
-  render() {
-    const metadata = get(this.props.uploadMetadata, `query.pages[${this.props.upload.id}]`);
-    const imageDescription = get(metadata, 'imageinfo[0].extmetadata.ImageDescription.value');
-    const width = get(metadata, 'imageinfo[0].width');
-    const height = get(metadata, 'imageinfo[0].height');
-
-    let size = get(metadata, 'imageinfo[0].size');
-    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
-    const i = Math.floor(Math.log(size) / Math.log(1024));
-    size = `${parseFloat((size / (1024 ** i)).toFixed(2))} ${sizes[i]}`;
-
-    const imageUrl = get(metadata, 'imageinfo[0].url');
+  }
 
-    const profileLink = `/users/${encodeURIComponent(this.props.upload.uploader)}`;
-    const author = <a href={profileLink} target="_blank">{this.props.upload.uploader}</a>;
-    const source = get(metadata, 'imageinfo[0].extmetadata.Credit.value');
-    const license = get(metadata, 'imageinfo[0].extmetadata.LicenseShortName.value');
-    const globalUsage = get(metadata, 'globalusage', []);
-    let usageTableElements;
-    if (globalUsage && (this.props.pageViews !== undefined)) {
-      usageTableElements = globalUsage.map((usage, index) => {
-          return (
-            <tr className="view-file-details" key={usage.url}>
-              <td className="row-details">{usage.wiki}&nbsp;&nbsp;&nbsp;</td>
-              <td className="row-details"><a href={usage.url}>{usage.title}</a>&nbsp;&nbsp;&nbsp;</td>
-              <td className="text-right row-details">{this.props.pageViews[index]}</td>
+  let fileUsageTable;
+  if (globalUsage.length > 0) {
+    fileUsageTable = (
+      <div>
+        <h1>{'\n'}</h1>
+        <h4>{I18n.t('uploads.file_usage')}</h4>
+        <table border="1">
+          <thead>
+            <tr>
+              <th>{I18n.t('uploads.wiki_big')}</th>
+              <th>{I18n.t('uploads.article_name')}</th>
+              <th>{I18n.t('uploads.views_per_day')}</th>
             </tr>
-          );
-      });
-    }
+          </thead>
+          <tbody>
+            {usageTableElements}
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+  let categoriesList = [];
+  let categories;
+  forEach(get(metadata, 'categories', []), (category) => {
```

Comment:
I think we can slowly migrate away from using lodash functions for things which we can natively do in JS.

```
forEach(get(metadata, 'categories', []), (category) => {
```

would be just

```
(metadata.categories ?? []).forEach((category) => {
```

Reply:
sure